src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeLibrary.cs

```diff
@@ -256,5 +257,24 @@ public static void SetDllImportResolver(Assembly assembly, DllImportResolver res
 
             return resolver(libraryName, assembly, hasDllImportSearchPathFlags ? (DllImportSearchPath?)dllImportSearchPathFlags : null);
         }
+
+        [DllImport(RuntimeHelpers.QCall, SetLastError = true)]
+        private static extern IntPtr GetEntryPointModuleHandleInternal();
+
+        /// <summary>
+        /// Get a handle that can be used with <see cref="GetExport" /> or <see cref="TryGetExport" /> to resolve exports from the entry point module.
+        /// </summary>
+        /// <returns> The handle that can be used to resolve exports from the entry point module.</returns>
+        public static IntPtr GetEntryPointModuleHandle()
+        {
+            IntPtr result = GetEntryPointModuleHandleInternal();
+            // I don't know when a failure case can occur here, but checking for it and throwing an exception
+            // if we encounter it.
+            if (result == IntPtr.Zero)
+            {
+                Marshal.ThrowExceptionForHR(Marshal.GetLastPInvokeError());
```

Comment:
Is the error returned here really HResult?

Reply:
I don't think so. It would typically be a Windows error code or an errno value. Which means it wouldn't normally have the error bit set.